README.md

```diff
@@ -1,27 +1,305 @@
+Install the package with:
 
-```code
+```bash
 pip install descope
-export DESCOPE_PROJECT_ID=<ProjectID>
 ```
 
-## What do you want to implement?
+## Setup
+
+A Descope `Project ID` is required to initialize the SDK. Find it on the
+[project page in the Descope Console](https://app.descope.com/settings/project).
+
+```python
+from descope import DescopeClient
+
+# Initialized after setting the DESCOPE_PROJECT_ID env var
+descope_client = DescopeClient()
+
+# ** Or directely **
+descope_client = DescopeClient(project_id="<Project ID>")
+```
+
+## Usage
+
+Here are some examples how to manage and authenticate users:
+
+### OTP Authentication
+
+Send a user a one-time password (OTP) using your preferred delivery method (_email, SMS, Whatsapp message_). An email address or phone number must be provided accordingly.
+
+The user can either `sign up`, `sign in` or `sign up or in`
+
+```python
+from descope import DeliveryMethod
+
+# Every user must have an identifier. All other user information is optional
+email = "desmond@descope.com"
+user = {"name": "Desmond Copeland", "phone": "212-555-1234", "email": email}
+descope_client.otp.sign_up(method=DeliveryMethod.EMAIL, identifier=email, user=user)
+```
+
+The user will receive a code using the selected delivery method. Verify that code using:
+
+```python
+jwt_response = descope_client.otp.verify_code(
+    method=DeliveryMethod.EMAIL, identifier=email, code=value
+)
+```
+
+The session and refresh JWTs should be returned to the caller, and passed with every request in the session. Read more on [session validation](#session-validation)
+
+### Magic Link
+
+Send a user a Magic Link using your preferred delivery method (_email, SMS, Whatsapp message_).
+The Magic Link will redirect the user to page where the its token needs to be verified.
+This redirection can be configured in code, or generally in the [Descope Console](https://app.descope.com/settings/authentication/magiclink)
+
+The user can either `sign up`, `sign in` or `sign up or in`
+
+```python
+from descope import DeliveryMethod
+
+descope_client.magiclink.sign_up_or_in(
+    method=DeliveryMethod.EMAIL,
+    identifier="desmond@descope.com",
+    uri="http://myapp.com/verify-magic-link", # Set redirect URI here or via console
+)
+```
+
+To verify a magic link, your redirect page must call the validation function on the token (`t`) parameter (`https://your-redirect-address.com/verify?t=<token>`):
+
+```python
+jwt_response = descope_client.magiclink.verify(token=token)
+```
+
+The session and refresh JWTs should be returned to the caller, and passed with every request in the session. Read more on [session validation](#session-validation)
+
+### Session Validation
+
+Every secure request performed between your client and server needs to be validated. The client sends
+the session and refresh tokens with every request, and they are validated using:
+
+```python
+descope_client.validate_session_request(session_token, refresh_token)
```

Comment:
maybe here mention the response you get from calling this ? what you get in case of success and failure ? 

also, I think refresh_token is not mandatory, so worth the mention